api/resolvers/blockHeight.js

```diff
@@ -0,0 +1,35 @@
+const cache = new Map()
+const expiresIn = 1000 * 60 // in milliseconds
+
+async function fetchBlockHeight () {
+  const url = 'https://mempool.space/api/blocks/tip/height'
+  const blockHeight = await fetch(url)
+    .then(res => res.text())
+    .then((body) => Number(body))
+    .catch((err) => {
+      console.error('blockHeight', err)
+      return 0
+    })
+  cache.set('block', { height: blockHeight, createdAt: Date.now() })
+  return blockHeight
+}
+
+async function getBlockHeight () {
+  if (cache.has('block')) {
+    const { height, createdAt } = cache.get('block')
+    const expired = createdAt + expiresIn < Date.now()
+    if (expired) fetchBlockHeight().catch(console.error) // update cache
+    return height // serve stale block height (this on the SSR critical path)
```

Comment:
This is basically the [`stale-while-revalidate` caching strategy](https://developer.chrome.com/docs/workbox/caching-strategies-overview/#stale-while-revalidate), right?

Reply:
Yea, pretty much. I just copied it from `api/resolvers/price.js`, so I can't take credit :)

Reply:
That was one of ekzyis' first contributions :)

Reply:
That's right but I looked at `git blame` and you updated the code to use that better caching strategy :)

See 3c8ea0db